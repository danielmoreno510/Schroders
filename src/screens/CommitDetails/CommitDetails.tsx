import React from 'react';
import {Text, View} from 'react-native';

import styles from './styles';
import {formatDate} from '../../utils/formUtils';

const RepositoryDetails: React.FC<{commit: ICommit}> = ({commit}) => (
  <View style={styles.container}>
    <Text style={styles.text}>Title: {commit.title}</Text>
    <Text style={styles.text}>body: {commit.body}</Text>
    <Text style={styles.text}>Author: {commit.user.login}</Text>
    <Text style={styles.text}>Date: {formatDate(commit.created_at)}</Text>
  </View>
);

export default RepositoryDetails;
