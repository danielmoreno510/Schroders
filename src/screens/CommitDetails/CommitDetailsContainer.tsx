import CommiDetails from './CommitDetails';
import {StoreService} from '../../redux/StoreService';

export const mapStateToProps = (state: ICommitListStateProps) => ({
  commit: state.commits.commit,
});

const CommitDetailsContainer =
  StoreService.connect(mapStateToProps)(CommiDetails);

export default CommitDetailsContainer;
