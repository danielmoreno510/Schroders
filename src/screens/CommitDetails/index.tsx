import CommitDetailsContainer from './CommitDetailsContainer';

export default CommitDetailsContainer;
