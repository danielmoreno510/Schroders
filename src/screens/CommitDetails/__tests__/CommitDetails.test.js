import React from 'react';
import {shallow} from 'enzyme';

import CommitDetails from '../CommitDetails';

describe('CommitDetails component', () => {
  describe('GIVEN a set of valid props', () => {
    const testProps = {
      commit: {user: {login: ''}, created_at: new Date()},
    };

    describe('WHEN the component is rendered', () => {
      const shallowWrapper = shallow(<CommitDetails {...testProps} />);

      it('THEN should display a regular CommitDetails', () => {
        expect(shallowWrapper.debug()).toMatchSnapshot();
      });
    });
  });
});
