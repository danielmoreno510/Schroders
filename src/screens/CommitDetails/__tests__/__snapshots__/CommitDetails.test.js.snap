// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommitDetails component GIVEN a set of valid props WHEN the component is rendered THEN should display a regular CommitDetails 1`] = `
"<View style={{...}}>
  <Text style={{...}}>
    Title: 
  </Text>
  <Text style={{...}}>
    body: 
  </Text>
  <Text style={{...}}>
    Author: 
  </Text>
  <Text style={{...}}>
    Date: 
    2023-09-25
  </Text>
</View>"
`;
